python/vegafusion/vegafusion/datasource/dfi_datasource.py

```diff
@@ -0,0 +1,85 @@
+from typing import Iterable
+import re
+import pyarrow as pa
+from pyarrow.interchange import from_dataframe
+
+from typing import Any
+from ._dfi_types import DtypeKind, DataFrame as DfiDataFrame
+from .datasource import Datasource
+
+# Taken from private pyarrow utilities
```

Comment:
I copied some private utilities from pyarrow that handle converting DataFrame Interchange Protocol types to pyarrow schema types.